packages/button/src/StyledButton/StyledButton.tsx

```diff
@@ -0,0 +1,164 @@
+import React from "react"
+import classnames from "classnames"
+import { OverrideClassName } from "@kaizen/component-base"
+import { Icon } from "@kaizen/component-library"
+import { LoadingSpinner } from "@kaizen/loading-spinner"
+import styles from "./StyledButton.module.scss"
+
+const WorkingContents = ({
+  contents,
+}: {
+  contents: React.ReactNode
+}): JSX.Element => (
+  <>
+    <span className={styles.hidden} aria-hidden="true">
+      {contents}
+    </span>
+    <LoadingSpinner
+      accessibilityLabel=""
+      size="sm"
+      classNameOverride={styles.loadingSpinner}
+    />
+  </>
+)
+
+interface ButtonIconProps {
+  icon?: React.SVGAttributes<SVGSymbolElement>
+  iconPosition?: "start" | "end"
+}
+
+interface ButtonContentsProps extends ButtonIconProps {
+  contents: React.ReactNode
+}
+
+const ButtonContents: React.VFC<ButtonContentsProps> = ({
+  contents,
+  icon,
+  iconPosition = "start",
+}) => {
+  if (!icon) return <>{contents}</>
+
+  return (
+    <span
+      className={classnames(
+        styles.buttonContents,
+        iconPosition === "end" && styles.iconPositionEnd
+      )}
+    >
+      <span className={styles.iconContainer}>
+        <Icon icon={icon} role="presentation" />
+      </span>
+      <span>{contents}</span>
+    </span>
+  )
+}
+
+interface BaseStyledButtonProps
+  extends ButtonIconProps,
+    OverrideClassName<unknown> {
+  variant: "default" | "primary" | "secondary" | "secondaryDestructive"
+  isReversed?: boolean
+  isWorking?: boolean
+  contentsPropName?: string
+  isDisabled?: boolean
+}
+
+export const getStyledButtonClassNames = ({
+  variant,
+  isReversed,
+  isWorking,
+  isDisabled,
+  classNameOverride,
+}: BaseStyledButtonProps): string =>
+  classnames(
+    styles.styledButton,
+    styles[variant],
+    isReversed && styles.isReversed,
+    isWorking && styles.isWorking,
+    isDisabled && styles.isDisabled,
+    classNameOverride
+  )
+
+// V1: Using children
+export interface StyledButtonProps extends BaseStyledButtonProps {
+  children: React.ReactElement
+}
+
+export const StyledButton: React.VFC<StyledButtonProps> = ({
+  children,
+  isWorking,
+  contentsPropName = "children",
```

Comment:
Because we construct the inner content (for the working state), we need to know where the contents go within the child. In most cases, it should be the children (eg. `<button>Contents</button>`), but if there's a custom component structure (eg. `<CustomComponent contents="Contents" />`, this can still work.

Reply:
Yes it will 😄